src/map/skill.c

```diff
@@ -12814,7 +12899,7 @@ int skill_unit_onleft(uint16 skill_id, struct block_list *bl, int64 tick) {
 		case PF_FOGWALL:
 			if (sce) {
 				status_change_end(bl, type, INVALID_TIMER);
-				if ((sce=sc->data[SC_BLIND])) {
+				if (sc != NULL && (sce = sc->data[SC_BLIND])) {
```

Comment:
I think a nullpo_retb() would be better, to make it clear the `sc` should never be NULL here (see `sce = (sc != NULL && type != SC_NONE) ? sc->data[type] : NULL;` at the beginning of the function